### ISSUE TITLE
Bug: TypeError: undefined is not an object (evaluating 'i[Symbol.iterator]')

### ISSUE THREAD

**Issue by paradox37** [PostHog/posthog-js#550]
Since updating to latest version, we are seeing a lot of errors coming from PostHog on Sentry:

```
TypeError: undefined is not an object (evaluating 'i[Symbol.iterator]')
  at _createForOfIteratorHelper(../../../../node_modules/posthog-js/dist/es.js:272:45)
  at <object>.value(../../../../node_modules/posthog-js/dist/es.js:4341:25)
  at <anonymous>(../../../../node_modules/posthog-js/dist/es.js:4260:19)
  at forEach([native code])
  at this.observer(../../../../node_modules/posthog-js/dist/es.js:4259:29)
  at observe([native code])
  at <anonymous>(../../../../node_modules/posthog-js/dist/es.js:4269:109)
  at forEach([native code])
  at <object>.value(../../../../node_modules/posthog-js/dist/es.js:4266:20)
  at <object>.value(../../../../node_modules/posthog-js/dist/es.js:4305:14)
  at <object>.value(../../../../node_modules/posthog-js/dist/es.js:4446:140)
  at req.onreadystatechange(../../../../node_modules/posthog-js/dist/es.js:5835:11)
  at sentryWrapped(../../../../../node_modules/@sentry/browser/esm/helpers.js:90:17)
```

Previously, we were running ^1.34.0. Then we updated to ^1.43.1. After that, error started to appear. Then we updated to ^1..45.1, but it didn't help, it's still happening. It seems it's coming during recording sessions. 1.46.1 changelog does not indicate that anything like this is fixed.

Can this be fixed with some priority? We keep getting errors very often from our production and it's not looking good.

**Comment by pauldambra** [comment]
Hey @paradox37 👋 

1.46.1 removes Sentry types from posthog-js and since the first line in that error references the `@sentry` module. That might fix the problem for you. My guess is you have a different version of sentry than posthog and they're incompatible - that's why we removed them 😊 

Can you try that and let us know how it goes?

**Comment by paradox37** [comment]
Thanks. Will do. We are releasing new version of production on Monday, so I'll let you know after couple of days.

**Comment by rohitpaulk** [comment]
We're seeing this too, not sure if it is related to Sentry. We do use Sentry in production, but I tried disabling it locally and the error is present on Safari (but not Edge).

<img width="463" alt="Screenshot 2023-02-27 at 14 01 21" src="https://user-images.githubusercontent.com/3893573/221583867-790b5d39-0bce-45f6-b35a-47ce9717012b.png">

<img width="991" alt="Screenshot 2023-02-27 at 14 01 24" src="https://user-images.githubusercontent.com/3893573/221583863-6486316c-48cc-408b-b39a-b3c78203ed8e.png">

We're on 1.46.2.

**Comment by olliejm** [comment]
We have just started using posthog-js, on version 1.46.2, and also found that this problem is occurring on Safari. We don't use Sentry (at least not directly, and I don't see any result searching for `sentry` in our client app's node_modules either). It's a React based web application and this bug currently completely breaks rendering unless we skip posthog initialisation on Safari.

**Comment by pauldambra** [comment]
Thanks @rohitpaulk that's really useful context. I see the same error on safari too.

**Comment by pauldambra** [comment]
We've released 1.48.2 which fixes this error in local testing
